cmake/GetTriple.cmake

```diff
@@ -0,0 +1,89 @@
+# cmake/GetTriple.cmake
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+# Based on libcxx/cmake/Modules/GetTriple.cmake
+
+# Define functions to get the host and target triple.
+
+function(get_host_triple out out_arch out_vendor out_os)
+  # Get the architecture.
+  set( arch "${CMAKE_HOST_SYSTEM_PROCESSOR}" )
+  # i686 is an enhanced version of x86 
+  if ( arch STREQUAL "x86" )
+    set( arch "i686" )
+  endif ()
+
+  # Get the vendor.
+  if ( "${CMAKE_HOST_SYSTEM_NAME}" MATCHES "^Darwin.*" )
+    set( vendor "apple" )
+  else ()
+    set( vendor "pc" )
+  endif ()
+
+  # Get os.
+  if ( "${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows" )
+    set( os "win32" )
```

Comment:
I am not so sure how relevant this actually is, given that NEST is not officially supported on Windows ("we do not tests this regularly [building on Windows] and do not support NEST under Cygwin at present"), but the lines are only relevant, if the host system name is `Windows`, then this will default to `win32`. In other cases, the `CMAKE_HOST_SYSTEM_NAME` will just be cast to lower case. Maybe on a 64bit Windows this will default to `win64` - I have not tested this - or on MinGW this will default to `mingw32` (as the [autoconf counterpart](https://sourceforge.net/p/mingw-w64/wiki2/TypeTriplets/)).